src/System/include/BipedalLocomotion/System/AdvanceableRunner.h

```diff
@@ -314,16 +325,30 @@ AdvanceableRunner<_Advanceable>::run(std::optional<std::reference_wrapper<Barrie
             // set the output
             this->m_output->set(this->m_advanceable->getOutput());
 
-            // release the CPU
-            BipedalLocomotion::clock().yield();
-
             // this is enabled only if telemetry is set to true.
             if (m_isTelemetryEnabled && wakeUpTime < BipedalLocomotion::clock().now())
             {
                 std::lock_guard<std::mutex> guard(m_infoMutex);
                 m_info.deadlineMiss++;
+                if (m_maximumNumberOfAcceptedDeadlineMiss >= 0
+                    && m_info.deadlineMiss > m_maximumNumberOfAcceptedDeadlineMiss)
+                {
+                    log()->error("{} - {} Experienced {} deadline misses. The maximum accepted "
+                                 "number is {}.",
+                                 logPrefix,
+                                 m_info.name,
+                                 m_info.deadlineMiss,
+                                 m_maximumNumberOfAcceptedDeadlineMiss);
+
+                    // we have to close the runner
+                    this->m_isRunning = false;
+                    break;
```

Comment:
This is done only when telemetry is enabled, is this intentional? Or perhaps know that deadlineMiss is used for something else we should change the logic?

Reply:
> This is done only when telemetry is enabled, is this intentional

Before I used it just to print some info at the end of the application but probably now it's better to keep this always active. What do you think?